src/backend/distributed/commands/function.c

```diff
@@ -1373,27 +1393,29 @@ PostprocessCreateFunctionStmt(Node *node, const char *queryString)
 		return NIL;
 	}
 
-	ObjectAddress functionAddress = GetObjectAddressFromParseTree((Node *) stmt, false);
+	List *functionAddresses = GetObjectAddressFromParseTree((Node *) stmt, false);
+	Assert(list_length(functionAddresses) == 1);
 
-	if (IsObjectAddressOwnedByExtension(&functionAddress, NULL))
+	if (IsAnyObjectAddressOwnedByExtension(functionAddresses, NULL))
 	{
 		return NIL;
 	}
 
 	/* If the function has any unsupported dependency, create it locally */
-	DeferredErrorMessage *errMsg = DeferErrorIfHasUnsupportedDependency(&functionAddress);
+	DeferredErrorMessage *errMsg = DeferErrorIfAnyObjectHasUnsupportedDependency(
+		functionAddresses);
 
 	if (errMsg != NULL)
 	{
 		RaiseDeferredError(errMsg, WARNING);
 		return NIL;
 	}
 
-	EnsureDependenciesExistOnAllNodes(&functionAddress);
+	EnsureAllObjectDependenciesExistOnAllNodes(functionAddresses);
 
 	List *commands = list_make1(DISABLE_DDL_PROPAGATION);
-	commands = list_concat(commands, CreateFunctionDDLCommandsIdempotent(
-							   &functionAddress));
+	commands = list_concat(commands, CreateAllFunctionDDLCommandsIdempotent(
```

Comment:
Ok, I keep getting here. But, I'd rather drop `CreateAllFunctionDDLCommandsIdempotent` function, and do

```
/* we already asserted above that there is a single */
ObjectAddress *functionAddress = (ObjectAddress *) linitial(functionAddresses);

... and use CreateFunctionDDLCommandsIdempotent ... 
```


I suggest this because `CreateAllFunctionDDLCommandsIdempotent` is really not a generic function. We are sure that there is no caller that can call this with multiple objects.